.github/workflows/call_test_package.yml

```diff
@@ -0,0 +1,107 @@
+        required: true
+        type: boolean
+        default: false
+
+
+# disable all the permission for the workflow
+permissions: {}
+
+jobs:
+  build-test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true  # if on job fail all the other jobs are cancelled
+      matrix:
+        python-version: ["3.13", "3.12"] #CHECK: not forget to add new version of python
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    
+    # name for files
+    env:
+      REPORT_NAME: ${{ matrix.python-version }}-os-${{ matrix.os }}
+
+    steps:
+    # Checkout to the branch for getting the right code
+    - uses: actions/checkout@v4
+
+      # skip tests
+    - name: Skip the tests
+      if: ${{ inputs.skip_test }}
+      run: |
+        (echo '<html><body><p>NO TESTS</p></body></html>') > report-${{env.REPORT_NAME}}.html
+        touch log-${{env.REPORT_NAME}}.txt
+        mkdir cov
+        touch cov/cov-${{env.REPORT_NAME}}.xml
+
+    # Install python
+    - name: Set up Python ${{ matrix.python-version }}
+      if: ${{ ! inputs.skip_test }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    
+    # Install dependency for running the tests
+    - name: Install the dependance
+      if: ${{ !inputs.skip_test && !startsWith(matrix.os, 'windows') }}
+      run: |
+        python -m venv .venv
+        source .venv/bin/activate
+        pip install --upgrade uv pip
+        uv pip install -e ".[test]"
+    - name: Install the dependance Windows
+      if: ${{ !inputs.skip_test && startsWith(matrix.os, 'windows') }}
+      run: |
+        python -m venv .venv
+        .venv\Scripts\python.exe -m pip install --upgrade uv pip
+        dir .venv\Scripts\
+        .venv\Scripts\uv.exe pip install -e ".[test]"
+    
+    # Run the tests with coverage and report add this end
+    # option for the coverage: --cov=. --cov-report=xml:... 
+    # option for mardown report: --md-report --md-report-output= ...
+    # option for html report: --self-contained-html --html=report-...
+    # option for duration report: --pytest-durations= number of second --pytest-resultlog=log-...
+    # inline command for being compatible on linux and windows
+    - name: Run pytest
+      if: ${{ !inputs.skip_test && !startsWith(matrix.os, 'windows') }}
+      run: |
+          source .venv/bin/activate
+          python -m pytest --cov=. --cov-report=xml:cov/cov-${{env.REPORT_NAME}}.xml --self-contained-html --html=report-${{env.REPORT_NAME}}.html --pytest-durations=10 --pytest-resultlog=log-${{env.REPORT_NAME}}.txt
+    - name: Run pytest  Windows
+      if: ${{ !inputs.skip_test && startsWith(matrix.os, 'windows') }}
+      run: |
+          .venv\Scripts/python.exe -m pytest --cov=. --cov-report=xml:cov/cov-${{env.REPORT_NAME}}.xml --self-contained-html --html=report-${{env.REPORT_NAME}}.html --pytest-durations=10 --pytest-resultlog=log-${{env.REPORT_NAME}}.txt
```

Comment:
maybe in a follow up PR but you could this shorter by relying on tox to create the virtual environment and running the proper commands

this would also make it easier to run the same thing in CI and locally